ls1intum/Artemis
# `Development`: Upgrade from tslint to eslint

### Checklist
#### General



### Motivation and Context



TSLint is end-of-life since 2020, we still used it as a [dependency](https://github.com/ls1intum/Artemis/pull/5947/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L159).
See  also: https://github.com/palantir/tslint/issues/4534

### Description


Ran the upgrade steps and made sure that all linting errors are fixed or temporarily ignored.

### Steps for Testing


Make sure the code changes do not change the functionality and everything still works correctly.

### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test
